src/mixins/nr-affiliation-mixin.ts

```diff
@@ -1,68 +1,84 @@
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Mixins } from 'vue-property-decorator'
 import { Action } from 'vuex-class'
 import AuthServices from '@/services/auth.services'
 import BusinessServices from '@/services/business.services'
-import { BusinessRequest, CreateNRAffiliationRequestBody, NameRequestI } from '@/interfaces'
+import { BusinessRequest, NameRequestI } from '@/interfaces'
 import { ActionBindingIF } from '@/interfaces/store-interfaces'
 import { navigate } from '@/plugins'
+import { CommonMixin } from '@/mixins'
+import { CREATED, BAD_REQUEST } from 'http-status-codes'
 
 @Component({})
-export class NrAffiliationMixin extends Vue {
+export class NrAffiliationMixin extends Mixins(CommonMixin) {
   // Global action
   @Action setAffiliationErrorModalVisible!: ActionBindingIF
 
   /**
-   * Creates an affiliation between the current authenticated account and the current NR.
+   * Creates an affiliation between the current authenticated account and the current NR
+   * and then navigates to My Business Registry page.
    * @param nr the NR to affiliate
    */
   async createAffiliation (nr: NameRequestI): Promise<any> {
     try {
       // show spinner since the network calls below can take a few seconds
       this.$root.$emit('showSpinner', true)
 
-      const currentOrganizationId = JSON.parse(sessionStorage.getItem('CURRENT_ACCOUNT')).id
-      const requestBody: CreateNRAffiliationRequestBody = {
-        businessIdentifier: nr.nrNum,
-        phone: nr.applicants?.phoneNumber || '',
-        email: nr.applicants?.emailAddress || ''
+      // NB: fall back is user's default account
+      const accountId = JSON.parse(sessionStorage.getItem('CURRENT_ACCOUNT')).id || 0
+
+      //
+      // try to affiliate NR to current account
+      //
+      const createAffiliationResponse = await AuthServices.createNrAffiliation(accountId, nr)
+      if (
+        createAffiliationResponse.status === BAD_REQUEST &&
+        createAffiliationResponse.data.code === 'NR_CONSUMED'
```

Comment:
Actually, it's not necessarily consumed. It would be nice to know whether the NR is already affiliated, or consumed, or expired, or whatever. Then we could give the user some options (or instructions) here.

---

Actually, something needs to be done now (*): if I am logged in and create a NR then it is automatically affiliated to my account, so this call fails.

What should happen is that the UI moves to the next step (try to create business), and if the business is already created then navigate to open it.

(*) I am looking into what needs to be updated to do this. In the meantime, this does not block this PR -- it still works OK, ie:

![image](https://user-images.githubusercontent.com/10002889/183961761-302a3f68-7fe0-4d40-a06f-0fdb84525e7b.png)

Reply:
Code is updated. This conversation is obsolete.